agent/hibinit-agent

```diff
@@ -39,23 +39,42 @@ IMDS_API_TOKEN_PATH = 'latest/api/token'
 IMDS_SPOT_ACTION_PATH = 'latest/meta-data/hibernation/configured'
 MAX_SWAP_SIZE_OFFSET_ALLOWED = 100 * 1024
 
+# Sigterm_handler Behaviour Modifiers
+CRITICAL_PROCESS_IN_PROGRESS = False
+SHUTDOWN_REQUESTED = False
+
+
 def log(message):
     if log_to_syslog:
         syslog.syslog(message)
 
-def sigterm_handler(signal, frame):
-    #save the state here or do whatever you want
-    log('Process killed cleaning up!')
-    cmd = "swapon -s | cut -f1,1 | grep -w {name}"
-    cmd = cmd.format(name=SWAP_FILE)
-    swapoff = "swapoff {filename}"
-    cmd = check_output(cmd, shell=True)
-    if cmd.strip() == SWAP_FILE:
-        print("Turning off swap for cleanup")
-        swapoff = swapoff.format(filename=SWAP_FILE)
-        check_call(swapoff, shell=True)
-    if os.path.isfile(SWAP_FILE) and os.access(SWAP_FILE, os.R_OK):
-        os.remove(SWAP_FILE)
+
+def sigterm_handler(signum, frame):
+    if CRITICAL_PROCESS_IN_PROGRESS:
```

Comment:
nit: is CRITICAL_SECTION more apt?

Reply:
I suppose this is an alternative that does make sense.

However, I believe the current naming is better for legibility, especially with general python style guidelines.
(`If critical process in process` is closer to general speech, while `if critical section` is somewhat ambiguous)

Reply:
Okay when we say process, I thought it refers to a process running in the OS. And to my understanding the process priority itself doesnot change but the section of the code it is running is critical/non critical hence the suggestion. 

If process refers to something else in the current  context and if you believe critical process is apt, then I am fine with this as well.

Reply:
I suppose the most accurate naming would be `CRITICAL_SUBPROCESSES_IN_PROGRESS`. 

But as this is a nit, this will be sometime best tackled in the later planned refactoring. :)

Reply:
I know its just a name but I am still curious to know why you want this to be called a `SUBPROCESS`/`PROCESS`.

Reply:
Synced offline. The `update_kernel_swap_offset` also creates subprocess for various functionality using `check_call`  apart from executing code on the same process . While I still think `section/routine` would be more apt but I get why you used `process` as  the critical task are being done by the subprocess itself.